contracts/zero-ex/contracts/src/features/MetaTransactionsFeatureV2.sol

```diff
@@ -0,0 +1,620 @@
+// SPDX-License-Identifier: Apache-2.0
+/*
+  Copyright 2023 ZeroEx Intl.
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+pragma solidity ^0.6.5;
+pragma experimental ABIEncoderV2;
+
+import "@0x/contracts-erc20/src/IEtherToken.sol";
+import "@0x/contracts-utils/contracts/src/v06/errors/LibRichErrorsV06.sol";
+import "@0x/contracts-utils/contracts/src/v06/LibBytesV06.sol";
+import "@0x/contracts-utils/contracts/src/v06/LibSafeMathV06.sol";
+import "../errors/LibMetaTransactionsRichErrors.sol";
+import "../fixins/FixinCommon.sol";
+import "../fixins/FixinReentrancyGuard.sol";
+import "../fixins/FixinTokenSpender.sol";
+import "../fixins/FixinEIP712.sol";
+import "../migrations/LibMigrate.sol";
+import "../storage/LibMetaTransactionsStorage.sol";
+import "./interfaces/IFeature.sol";
+import "./interfaces/IMetaTransactionsFeatureV2.sol";
+import "./interfaces/IMultiplexFeature.sol";
+import "./interfaces/INativeOrdersFeature.sol";
+import "./interfaces/ITransformERC20Feature.sol";
+import "./libs/LibSignature.sol";
+
+/// @dev MetaTransactions feature.
+contract MetaTransactionsFeatureV2 is
+    IFeature,
+    IMetaTransactionsFeatureV2,
+    FixinCommon,
+    FixinReentrancyGuard,
+    FixinEIP712,
+    FixinTokenSpender
+{
+    using LibBytesV06 for bytes;
+    using LibRichErrorsV06 for bytes;
+
+    /// @dev Describes the state of a meta transaction.
+    struct ExecuteState {
+        // Sender of the meta-transaction.
+        address sender;

```

Comment:
Again, this is confusing terminology

Reply:
This terminology is consistent with meta-txn v1, though I agree, it is a bit confusing. The sender is the person submitting the meta-txn on someone else's behalf. The signer is the taker.